samples/AudioPlayerSample/ViewModels/AudioRecorderPageViewModel.cs

```diff
@@ -4,36 +4,117 @@ namespace AudioPlayerSample.ViewModels;
 
 public class AudioRecorderPageViewModel : BaseViewModel
 {
-	readonly IAudioManager audioManager;
-	IAudioRecorder? audioRecorder;
+   ContentPage page;
 
-	public AudioRecorderPageViewModel(
-		IAudioManager audioManager)
-	{
-		StartCommand = new Command(Start);
+   readonly IAudioManager audioManager;
+   IAudioRecorder audioRecorder;
+   IAudioPlayer audioPlayer;
+   IAudioSource audioSource = null;
 
-		this.audioManager = audioManager;
-	}
+   string recordButtonText = "Record";
+   public string RecordButtonText
+   {
+      get => recordButtonText;
+      set
+      {
+         recordButtonText = value;
+         NotifyPropertyChanged();
+      }
+   }
 
-	public Command StartCommand { get; set; }
+   Color recordButtonColor = Colors.Blue;
+   public Color RecordButtonColor
+   {
+      get => recordButtonColor;
+      set
+      {
+         recordButtonColor = value;
+         NotifyPropertyChanged();
+      }
+   }
 
-	async void Start()
-	{
-		audioRecorder = audioManager.CreateRecorder();
+   double audioTime = 0;
+   public double AudioTime
+   {
+      get => audioTime;
+      set
+      {
+         audioTime = value;
+         NotifyPropertyChanged();
+      }
+   }
 
-		try
-		{
-			await audioRecorder.StartAsync();
+   public AudioRecorderPageViewModel(IAudioManager audioManager)
+   {
+      StartCommand = new Command(Start);
+      PlayCommand = new Command(playAudio);
+      this.audioManager = audioManager;
+   }
 
-			await Task.Delay(5000);
+   public void setPage(ContentPage page)
+   {
+      try
+      {
+         this.page = page;
+      }
+      catch (Exception ex)
+      {
+         page.DisplayAlert("Alert", $"setPage Exception: {ex.Message}", "OK");
+      }
+   }
 
-			var audioSource = await audioRecorder.StopAsync();
+   void DonePlaying(object sender, EventArgs e)
+   {
+      AudioTime = audioPlayer.Duration;
+   }
+   
+   public Command PlayCommand { get; set; }
+   public async void playAudio()
+   {
+      if (audioSource != null)
+      {
+         audioPlayer = this.audioManager.CreatePlayer(((FileAudioSource)audioSource).GetAudioStream());
+         audioPlayer.PlaybackEnded += DonePlaying;
+         await Task.Run(() =>
+               {
+                  audioPlayer.Play();
+               });
+      }
+   }
 
-			this.audioManager.CreatePlayer(((FileAudioSource)audioSource).GetAudioStream()).Play();
-		}
-		catch (Exception ex)
-		{
+   public Command StartCommand { get; set; }
+   async void Start()
+   {
+      // This must be done for Android to avoid an exception but I don't think it would hurt in any case
+      if (await HavePermissionMicrophoneAsync())
```

Comment:
Perhaps we should wrap this up into something to demonstrate to consumers on how to check for permission or should we consider adding it into the recording implementation?